packages/lesswrong/components/messaging/NewConversationButton.tsx

```diff
@@ -31,7 +33,6 @@ const NewConversationButton = ({ user, currentUser, children, templateCommentId
     collectionName: "Conversations",
     fragmentName: 'conversationsListFragment',
     fetchPolicy: 'cache-and-network',
-    limit: 1,
```

Comment:
So my understanding is that this query would just get a single conversation, but since it doesn't take the `user._id` it just gets one conversation that includes the `currentUser`, which doesn't necessarily include the `user`.

I updated this to get all the `currentUser`'s conversations, then in `existingConversationCheck()` we look for the first conversation with the `user` and route to that. This way we don't keep creating new conversations between users.

Reply:
This query takes about four seconds to run for me in production using graphiql. I think it might be worth creating a new fragment to return only the participant ids since that's all you care about? Or even better would be to change the `userUntitledConversations` view to do all the querying on the database instead of doing some of it on the client.

For reference, here's the query I used:

```
query {
  conversations (input: {terms: {view: "userUntitledConversations", userId: "dGXDHWQ9CaLKHRx5Y"}}) {
    results {
_id
title
createdAt
latestActivity
participantIds
participants
archivedByIds
messageCount
    }
  }
}
```

Reply:
Thanks, yeah I think it's worth creating a new fragment for the `_id`s. I updated the view to take the other user's `_id` as well - is that what you were suggesting? Seems better now, let me know if you have other suggestions.

Reply:
I realized there was another bug that would catch conversations that included the current user and the target user, even if other users were in the conversation. So I updated the view to account for that.